### PR TITLE
chore: add slog http middleware

### PR DESCRIPTION
**Pull request by gjarmstrong**
## Linear Ticket Link

https://linear.app/underlineinfa/issue/SENG-804/create-http-middleware-using-slog

## Description

Add `slog` HTTP Middleware

## How did you test the changes?

- Created unit test

## Dependencies

N/A

### CODE REVIEW

**Review by josemarluedke — APPROVED** [review]

**Review by josemarluedke — APPROVED** [review]

**Review by caseyaokelly — APPROVED** [review]
LGTM 🎉

File: context.go [hunk 1]
```diff
@@ -2,8 +2,10 @@ package log
 
 import "context"
 
+type logContextType string
+
 // ContextKeyLogFields is the key for the logging fields context value.
-const ContextKeyLogFields = "nrfta/go-log/Fields"
+const ContextKeyLogFields logContextType = "nrfta/go-log/Fields"
```
Comment (gjarmstrong) [inline context.go:8]: I added the custom type here to better conform to the context package practices. It prefers a custom type so the risk of key collision is removed.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -14,13 +14,17 @@ jobs:
     name: Tests
     runs-on: ubuntu-latest
 
+    strategy:
+      matrix:
+        go-version: [1.14.x, 1.21.x]
```
Comment (m-bodmer) [inline .github/workflows/ci.yml:19]: What happens if you run the jobs with version `1.21.x` only? Whats stopping us from only using that version?

This library seems small enough that an upgrade from 1.14.x -> 1.21.x shouldn't be too time consuming.

Comment (gjarmstrong) [reply .github/workflows/ci.yml:19]: I'm not sure which services have all been upgraded yet, so I didn't want to cause any conflicts. I'm sure once we know all service that import this module are at or above 1.21 we can just update the module to that version.

**Review by m-bodmer — APPROVED** [review]

**Pull request merged**